edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/storage/MessageStore.cs

```diff
@@ -360,8 +362,11 @@ async Task<bool> DeleteMessageCallback(long offset, MessageRef messageRef)
                                 }
                             }
 
-                            // update Metrics for message counts
-                            Checkpointer.Metrics.SetQueueLength(await sequentialStore.Count(), endpointId, priority.ToString());
+                            // Since we will have the total *true* count of messages in the store, we need to set the counter here
+                            // to the total count of messages in the store.
+                            var count = await sequentialStore.Count();
+                            Checkpointer.Metrics.SetQueueLength(count, endpointId, priority);
```

Comment:
I can only assume this is somehow causing the weird behaviour in the graph I linked with this pr where we do not pick up on queue length where we left off... both master and release/1.2 work as expected... and there aren't too many other places in this code that have changed...